File: leetcode/12_sliding_window/643_Maximum_Average_Subarray_I.go
func findMaxAverage(nums []int, k int) float64 {
	sum := 0
	for _, num := range nums[:k] {
		sum += num
	}
	max_sum := sum
	for i := k; i < len(nums); i++ {
		sum -= nums[i-k]
		sum += nums[i]
		max_sum = max(sum, max_sum)
	}
	return float64(max_sum) / float64(k)
}

File: leetcode/12_sliding_window/643_Maximum_Average_Subarray_I.py
def findMaxAverage(nums: list[int], k: int) -> float:
    left, right = 0, k - 1
    curr_sum = sum(nums[left:k])
    max_sum = curr_sum

    while right < len(nums) - 1:
        curr_sum -= nums[left]
        left += 1

        right += 1
        curr_sum += nums[right]

        max_sum = max(curr_sum, max_sum)

    return max_sum / k


if __name__ == '__main__':
    print(findMaxAverage([1, 12, -5, -6, 50, 3], 4))
